app/backend/src/tests/test_search.py

```diff
@@ -158,6 +160,51 @@ def test_user_filter_meetup_status(db):
         assert [result.user.user_id for result in res.results] == [user_does_not_want_to_meet.id]
 
 
+def test_user_filter_language(db):
+    """
+    Test filtering users by language ability.
+    """
+    user_with_german_beginner, token11 = generate_user(hosting_status=HostingStatus.can_host)
+    user_with_japanese_conversational, token12 = generate_user(hosting_status=HostingStatus.can_host)
+
+    with session_scope() as session:
+        session.add(
+            LanguageAbility(user_id=user_with_german_beginner.id, language_code="deu", fluency=LanguageFluency.beginner)
+        )
+        session.add(
+            LanguageAbility(
+                user_id=user_with_japanese_conversational.id,
+                language_code="jpn",
+                fluency=LanguageFluency.fluent,
+            )
+        )
+
+    with search_session(token11) as api:
+        res = api.UserSearch(
+            search_pb2.UserSearchReq(
+                language_ability_filter=[
+                    api_pb2.LanguageAbility(
+                        code="deu",
+                        fluency=api_pb2.LanguageAbility.Fluency.FLUENCY_CONVERSATIONAL,
+                    )
+                ]
+            )
+        )
+        assert [result.user.user_id for result in res.results] != [user_with_german_beginner.id]
```

Comment:
you wrote this part few weeks ago - If you'd like, it'd be good to have a test where the user is, say beginner in a language but you search for conversational and they wouldn't show up.

Reply:
Yes I think the test is good: I just meant that in a test it’s better to test that the list is exactly what you expect, not that individual properties. If I understand correctly, the search should output an empty list since that user isn’t returned, right?